lib/nerves_ssh/options.ex

```diff
@@ -224,8 +224,13 @@ defmodule NervesSSH.Options do
     ]
   end
 
-  defp shell_opts(%{shell: :elixir, iex_opts: iex_opts}),
-    do: [{:shell, {Elixir.IEx, :start, [iex_opts]}}]
+  if Version.match?(System.version(), ">= 1.17.0-rc.0") do
+    defp shell_opts(%{shell: :elixir, iex_opts: iex_opts}),
+      do: [{:shell, {:iex, :start, [{:elixir_utils, :noop, []} | iex_opts]}}]
```

Comment:
I'm unsure this will work because iex.erl requires `:iex.start(Opts, Mfa)` for the arguments.

Have you tried it?

Also, `elixir_utils` technically worked for me, but am wondering if it needs to be different

Reply:
I didn't try. But yes, my change definitely won't work. I'll change back when I'm back at my computer.

Reply:
Updated. I still didn't test, but it matches your original call now.